community/io/src/main/java/org/neo4j/io/fs/FileUtils.java

```diff
@@ -340,6 +343,86 @@ public static File path( File root, String... path )
     }
 
     /**
+     * Attempts to discern if the given path is mounted on a device that can likely sustain a very high IO throughput.
+     * <p>
+     * A high IO device is expected to have negligible seek time, if any, and be able to service multiple IO requests
+     * in parallel.
+     *
+     * @param pathOnDevice Any path, hypothetical or real, that once fully resolved, would exist on a storage device
+     * that either supports high IO, or not.
+     * @param defaultHunch The default hunch for whether the device supports high IO or not. This will be returned if
+     * we otherwise have no clue about the nature of the storage device.
+     * @return Our best-effort estimate for whether or not this device supports a high IO workload.
+     */
+    public static boolean highIODevice( Path pathOnDevice, boolean defaultHunch )
+    {
+        // This method has been manually tested and correctly identifies the high IO volumes on our test servers.
+        if ( SystemUtils.IS_OS_MAC )
+        {
+            // Most macs have flash storage, so let's assume true for them.
+            return true;
+        }
+
+        if ( SystemUtils.IS_OS_LINUX )
+        {
+            try
+            {
+                FileStore fileStore = Files.getFileStore( pathOnDevice );
+                String name = fileStore.name();
+                if ( name.equals( "tmpfs" ) || name.equals( "hugetlbfs" ) )
```

Comment:
I'm not a fan of that aesthetic, and it's no change to behaviour since without a name we have nothing to go by.